src/HydrationMiddleware/HydratePublicProperties.php

```diff
@@ -2,13 +2,17 @@
 
 namespace Livewire\HydrationMiddleware;
 
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Collection;
 use Livewire\Exceptions\PublicPropertyTypeNotAllowedException;
 
 class HydratePublicProperties implements HydrationMiddleware
 {
     public static function hydrate($unHydratedInstance, $request)
     {
-        $publicProperties = $request->memo['data'];
+        $publicProperties = static::removeTypedProperties(
```

Comment:
Everything is looking good. My only thought is, is it possible to not have the "model property hydration" concerns leaked to other hydration middleware.

For example, we have to remove them here because we already dehydrated them earlier. Does it make sense to colocate the removal of the properties with the other middleware?

Reply:
I think we'd have to mess with the hydration middleware order if we wanted to do that. It's just a trade-off re: which is preferable.

Reply:
I guess we could call `HydrateEloquentModelsAsPublicProperties` before `HydratePublicProperties` and then remove the data associated with Eloquent models from the data array…